nestkernel/conn_builder.cpp

```diff
@@ -42,12 +43,95 @@
 // Includes from C++:
 #include <algorithm>
 
-nest::ConnBuilder::ConnBuilder( NodeCollectionPTR sources,
+
+nest::ConnBuilder::ConnBuilder( const std::string& primary_rule,
+  NodeCollectionPTR sources,
   NodeCollectionPTR targets,
   const DictionaryDatum& conn_spec,
   const std::vector< DictionaryDatum >& syn_specs )
+  : third_in_builder_( nullptr )
+  , third_out_builder_( nullptr )
+  , primary_builder_( kernel().connection_manager.get_conn_builder( primary_rule,
+      sources,
+      targets,
+      third_out_builder_,
+      conn_spec,
+      syn_specs ) )
+{
+}
+
+nest::ConnBuilder::ConnBuilder( const std::string& primary_rule,
+  const std::string& third_rule,
+  NodeCollectionPTR sources,
+  NodeCollectionPTR targets,
+  NodeCollectionPTR third,
+  const DictionaryDatum& conn_spec,
+  const DictionaryDatum& third_conn_spec,
+  const std::map< Name, std::vector< DictionaryDatum > >& syn_specs )
+  : third_in_builder_( new ThirdInBuilder( sources,
+    third,
+    third_conn_spec,
+    const_cast< std::map< Name, std::vector< DictionaryDatum > >& >( syn_specs )[ names::third_in ] ) )
+  , third_out_builder_( kernel().connection_manager.get_third_conn_builder( third_rule,
+      third,
+      targets,
+      third_in_builder_,
+      third_conn_spec,
+      // const_cast here seems required, clang complains otherwise; try to clean up when Datums disappear
+      const_cast< std::map< Name, std::vector< DictionaryDatum > >& >( syn_specs )[ names::third_out ] ) )
+  , primary_builder_( kernel().connection_manager.get_conn_builder( primary_rule,
+      sources,
+      targets,
+      third_out_builder_,
+      conn_spec,
+      const_cast< std::map< Name, std::vector< DictionaryDatum > >& >( syn_specs )[ names::primary ] ) )
+{
+}
+
+nest::ConnBuilder::~ConnBuilder()
+{
+  delete primary_builder_; // fully constructed CB has one
+
+  if ( third_in_builder_ )
```

Comment:
Just a small comment: delete should be safe even on nullptrs.

Reply:
OK, I will remove the ifs.